pkg/sql/distsqlrun/backfiller.go

```diff
@@ -13,21 +13,144 @@
 // permissions and limitations under the License.
 //
 // Author: Vivek Menezes (vivek@cockroachlabs.com)
+// Author: Jordan Lewis (jordan@cockroachlabs.com)
 
 package distsqlrun
 
 import (
 	"golang.org/x/net/context"
 
+	"sync"
+
+	"fmt"
+
 	"github.com/cockroachdb/cockroach/pkg/internal/client"
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
 	"github.com/cockroachdb/cockroach/pkg/sql/sqlbase"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/cockroach/pkg/util/protoutil"
+	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
 	"github.com/cockroachdb/cockroach/pkg/util/tracing"
 	"github.com/pkg/errors"
 )
 
+type backfiller interface {
+	// runChunk returns the next-key and an error. next-key is nil
+	// once the backfill is complete.
+	runChunk(
+		ctx context.Context,
+		mutations []sqlbase.DescriptorMutation,
+		span roachpb.Span,
+		chunkSize int64,
+	) (roachpb.Key, error)
+
+	// mutationFilter returns true if the mutation should be processed by the
+	// backfiller.
+	mutationFilter(sqlbase.DescriptorMutation) bool
+
+	// name returns the name of the kind of entity this backfiller processes.
+	name() string
+}
+
+// abstractBackfiller is a processor that implements a distributed backfill of
+// an entity, like indexes or columns, during a schema change.
+type abstractBackfiller struct {
```

Comment:
I think I'd call this struct backfiller to represent either of the two backfillers.

and I'd call the backfiller interface runChunkInterface or applyBackfillForChunk or something equivalent. I think name() can just be a string added to backfiller? And if you want mutationFilter can just a function passed in. And if it works runChunk could also very well be a function passed in? Doesn't look like either of these two functions review state or have any side effects.